### PR TITLE
feat(settings): add occ commands to handle admin delegation

### PR DESCRIPTION
**Pull request by Altahrim**
## Summary
Add commands to handle admin delegation settings: 
  - `settings:admin-delegation:show`: show current delegation
  - `settings:admin-delegation:add`: add a delegation to a group
  - `settings:admin-delegation:remove`: remove a delegation from a group 

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/settings/lib/Command/AdminDelegation/Add.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+		private AuthorizedGroupService $authorizedGroupService,
+		private IGroupManager $groupManager,
+	) {
+		parent::__construct();
+	}
+
+	protected function configure(): void {
+		$this
+			->setName('settings:admin-delegation:add')
+			->setDescription('add delegated settings')
```
Comment (artonge) [inline apps/settings/lib/Command/AdminDelegation/Add.php:50]:
```suggestion
			->setDescription('add setting delegation to a group')
```

**Review by artonge — CHANGES_REQUESTED** [review]

File: apps/settings/lib/Command/AdminDelegation/Add.php [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+		private IManager $settingManager,
+		private AuthorizedGroupService $authorizedGroupService,
+		private IGroupManager $groupManager,
+	) {
+		parent::__construct();
+	}
+
+	protected function configure(): void {
+		$this
+			->setName('settings:admin-delegation:add')
```
Comment (Pytal) [inline apps/settings/lib/Command/AdminDelegation/Add.php:49]:
```suggestion
			->setName('admin-delegation:add')
```

File: apps/settings/lib/Command/AdminDelegation/Remove.php [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+		private IManager $settingManager,
+		private AuthorizedGroupService $authorizedGroupService,
+		private IGroupManager $groupManager,
+	) {
+		parent::__construct();
+	}
+
+	protected function configure(): void {
+		$this
+			->setName('settings:admin-delegation:remove')
```
Comment (Pytal) [inline apps/settings/lib/Command/AdminDelegation/Remove.php:48]:
```suggestion
			->setName('admin-delegation:remove')
```

File: apps/settings/lib/Command/AdminDelegation/Show.php [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+	public function __construct(
+		private IManager $settingManager,
+		private AuthorizedGroupService $authorizedGroupService,
+	) {
+		parent::__construct();
+	}
+
+	protected function configure(): void {
+		$this
+			->setName('settings:admin-delegation:show')
```
Comment (Pytal) [inline apps/settings/lib/Command/AdminDelegation/Show.php:47]:
```suggestion
			->setName('admin-delegation:show')
```

File: apps/settings/lib/Command/AdminDelegation/Add.php [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+	}
+
+	/**
+	 * @param string $argumentName
+	 * @param CompletionContext $context
+	 * @return string[]
+	 */
+	public function completeArgumentValues($argumentName, CompletionContext $context): array {
+		return [];
+	}
```
Comment (Pytal) [inline apps/settings/lib/Command/AdminDelegation/Add.php:84]:
```suggestion
```
Seens identical to the Base class method?

File: apps/settings/lib/Command/AdminDelegation/Remove.php [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+	}
+
+	/**
+	 * @param string $argumentName
+	 * @param CompletionContext $context
+	 * @return string[]
+	 */
+	public function completeArgumentValues($argumentName, CompletionContext $context): array {
+		return [];
+	}
```
Comment (Pytal) [inline apps/settings/lib/Command/AdminDelegation/Remove.php:80]:
```suggestion
```

File: apps/settings/lib/Command/AdminDelegation/Show.php [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+	}
+
+	/**
+	 * @param string $argumentName
+	 * @param CompletionContext $context
+	 * @return string[]
+	 */
+	public function completeArgumentValues($argumentName, CompletionContext $context): array {
+		return [];
+	}
```
Comment (Pytal) [inline apps/settings/lib/Command/AdminDelegation/Show.php:104]:
```suggestion
```

File: apps/settings/lib/Command/AdminDelegation/Show.php [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+			->setDescription('show delegated settings')
+		;
+	}
+
+	protected function execute(InputInterface $input, OutputInterface $output): int {
+		$io = new SymfonyStyle($input, $output);
+		$io->title('Current delegations');
+
+		$adminSections = $this->settingManager->getAdminSections();
+		$sections = $this->settingManager->getAdminSections();
```
Comment (come-nc) [inline apps/settings/lib/Command/AdminDelegation/Show.php:57]:
```suggestion
		$sections = $this->settingManager->getAdminSections();
```
Seems unused?

Comment (Altahrim) [reply apps/settings/lib/Command/AdminDelegation/Show.php:57]: Completely!

File: apps/settings/lib/Command/AdminDelegation/Add.php (hunk 5 above)
Comment (Altahrim) [reply apps/settings/lib/Command/AdminDelegation/Add.php]: Yes, I wasn't sure and forgot to check… Thanks!

**Review by come-nc — APPROVED** [review]
Tested and works

**Review by Pytal — APPROVED** [review]

**Review by artonge — APPROVED** [review]

**Pull request merged**